IEnumerable feat: Add Services/Storages separations

aokigahara59 opened the pull request.

aokigahara59 committed:
IEnumerable feat: Add Services/Storages separations

Dem-Ivan commented on Services.Tests/EmployeeServiceTests.cs:
```diff
@@ -33,5 +44,66 @@ public void AddEmployeeArgumentNullException()
+                }
+                catch (Exception ex)
+                {
+                    _testOutputHelper.WriteLine("");
```
Зачем эта строка? Какую функцию она выполняет?

Dem-Ivan commented on Services.Tests/EmployeeServiceTests.cs:
```diff
@@ -33,5 +44,66 @@ public void AddEmployeeArgumentNullException()
+            }
+
+            var filteredEmployees = employeeService.GetEmployees(filter);
+            _testOutputHelper.WriteLine(filteredEmployees.Count().ToString());
```
Для чего это строка? Какую функцию она выполняет?

Dem-Ivan commented on Services/ClientService.cs:
```diff
@@ -1,11 +1,17 @@
+        private ClientStorage _clientStorage;
+        public ClientService(ClientStorage storage)
+        {
+            this._clientStorage = storage;
```
this - лишнее

Dem-Ivan commented on Services.Tests/EmployeeServiceTests.cs:
```diff
@@ -33,5 +44,66 @@ public void AddEmployeeArgumentNullException()
+            foreach (var employee in employees)
+            {
+                try
+                {
```
Ваш тест должен проверять метод получения сотрудников по фильтру. Если тут вывалится исключение как будут разворачиваться события?

aokigahara59 committed:
TaskIEnumerable-2 fix: fixed exceptions in get clients and employees tests

Fixed exceptions in get clients and employees by filter tests
If data incorrect test fall

aokigahara59 committed:
TaskIEnumerable-3 fix: fixed output in get client test
